Our custom footer component CSS

mmmavis opened the pull request:
Fixes #92

mmmavis commented on README.md:
```diff
@@ -31,7 +31,8 @@ dest/ <- Contains compiled code. Don't edit anything in this folder!
 └── scss/
-    ├── custom/ <- SCSS modules for non-Bootstrap components.
+    ├── custom/ <- SCSS for non-Bootstrap modules.
+    |     └── components/ <- SCSS for custom components.
```
Just wanna double - does it make sense to have the custom components dir live here? @gvn @alanmoo

gvn replied:
Yeah that seems good to me.

mmmavis commented:
Note that I've ngrok'ed it and got @kristinashu 's okay on the design.

gvn commented on src/scss/custom/components/_footer.scss:
```diff
@@ -0,0 +1,157 @@
+  list-style-type: none;
+  padding-left: 0;
+  margin-bottom: 0;
+
```
Is there a reason you didn't use flex layout for this?

mmmavis replied:
Um we can't anticipate what and how many footer links are gonna be there. With flexbox we can wrap links into multilines but we can't have space around each links equally set. 

```
  [ item 1 ]     [ item 2 item 2 item 2 ]    [ item 3 ]
    [ item 4 item 4 item 4 item 4 ] [ item 5 item 5 ] 
[ item 6  ] [ item 7 ] [ item 8 item 8 item 8 ] [ item 9 ]

```

alanmoo replied:
This functionality is possible with the grid in Bootstrap v4.alpha6. I'll work on a PR now to bump the version we're using to that so that this can be done with less CSS.

gvn replied:
Also @mmmavis this is a great guide if you're not already aware of it:

https://css-tricks.com/snippets/css/a-guide-to-flexbox

gvn commented on src/scss/custom/components/_footer.scss:
```diff
@@ -0,0 +1,157 @@
+        border-left: 1px solid transparentize($white,0.5);
+        top: 15%;
+        left: 0;
+      }   
```
I'm seeing trailing whitespace here. Can you set your editor to delete that on save?

mmmavis replied:
wait where? on line 148? I don't see a trailing space there 👀

gvn replied:
<img width="456" alt="screen shot 2017-01-23 at 9 45 31 am" src="https://cloud.githubusercontent.com/assets/206379/22215587/c491d92e-e150-11e6-8ce9-4d54f3bdc3d1.png">

gvn replied:
You use Sublime, right? There should be a plugin to use the `.editorconfig` file in this repo.

http://editorconfig.org

gvn commented on src/index.pug:
```diff
@@ -225,3 +225,31 @@ html
+      .container
+        ul.footer-links 
+          li 
+            a(href="#" class="info") Info
```
Can you namespace these classes. There's a pretty good chance for a conflict without one when this footer is added to other sites. Something like `footer-link-info`, `footer-link-email`, etc.

alanmoo commented:
@mmmavis I've updated MoFo-Bootstrap to include the newest grid stuff- rebase off master so you can take advantage of [these classes](https://v4-alpha.getbootstrap.com/layout/grid/#variable-width-content). If you don't get to look at these today, I'll throw together an example tomorrow morning.

mmmavis commented:
@alanmoo @gvn

I updated the PR with new Mozilla logo & some adjustments on the `.org-info` section. Reason for that is sometimes we will have more than 1 org on the footer (see [example](https://redpen.io/vafce1db0b32ec575b)) and we need a way to adjust logo size so one doesn't look too prominent. I added the NSF logo in this PR just to show you the why I'm making these adjustment. If the updated code looks okay I'll remove the NSF chunk and we can merge this PRRRRRRRRRR.

gvn commented:
This looks great, but we should take out the NSF stuff before landing because that's only for 1 site and `mofo-bootstrap` is just our shared code.

mmmavis commented:
NSF chunk removed.

gvn commented on src/scss/custom/components/_footer.scss:
```diff
@@ -0,0 +1,164 @@
+footer {
```
I think we should use a class instead of putting style directly on all `footer` elements. Any site that uses this library will have to override these rules for other uses of `footer`.

mmmavis replied:
I'm not super sure about this one. I guess it depends on if we are planning to force all our MoFo sites use the same footer (styling-wise) or not. I ***think*** we should make this an unified standard and let exceptions override the rules if necessary. 

Thoughts? /cc @kristinashu @taisdesouzalessa @alanmoo @gvn

kristinashu replied:
Yeah, that should all have the same styling/layout.

gvn replied:
My point is that the `footer` element can be used for other footers than the overall page footer. You can have a `footer` in another part of the page as a semantic element within another block. I'm not suggesting we don't have a standard footer, but that we just put a class on this footer to differentiate it (eg `<footer class="mofo-footer">`).

mmmavis replied:
Ah I see. I have the tendency to use `<footer>` once and only as the page footer that's why my assumption was there would only be one `<footer>` in the DOM.

Should we just name this `page-footer`?

gvn replied:
Yeah, that's common, but it's totally valid to have many `footer` elements throughout a page.

`page-footer` sounds like a good classname to me. 👍

alanmoo replied:
Agreed 100% that we need a separate class, though FWIW I like `mofo-footer` for clarity.

gvn replied:
Yeah, actually now that we're adding non Bootstrap components here we should establish a namespace prefix for component classes. In mofo-ui it's `mui-`. 

`mofo-` seems fine to me. We should document it in the README as well.

mmmavis commented on src/scss/custom/components/_footer.scss:
```diff
@@ -0,0 +1,164 @@
+  }
+}
+
+.org-info {
```
should i make this `footer-org-info` instead?

mmmavis commented:
@gvn @alanmoo should i wrap everything in `_footer.scss` in `.mofo-footer { ... }` then?

gvn commented:
Yeah, that's what I'd do. It keeps the whole thing namespaced that way. 👍

mmmavis commented:
@gvn @alanmoo PR updated! And followup ticket created for have README updated: https://github.com/mozilla/mofo-bootstrap/issues/99

mmmavis commented:
FINALLY

🍷 🍾 💃 🎉 :man_dancing: 🍻

mmmavis commented:
Sorry forgot to squash before merging. I'll remember to do it next time.
